src/serialization/sb3.js

```diff
@@ -272,6 +319,28 @@ const compressInputTree = function (block, blocks) {
     return block;
 };
 
+/**
+ * Interpret a serialized block to retrieve extension & opcode information. We can't deserialize a block without
+ * knowing whether its extension offers custom deserialization for the block, but we can't know that without knowing
+ * the block's opcode. In the case of a primitive that means we need to at least partially deserialize the block. This
+ * function provides a sort of "minimal deserialization" in order to break that dependency loop. For convenience it
+ * also handles non-primitive serialized blocks.
+ * @param {object|array} blockJSON - a serialized block in either object or primitive (array) form.
+ * @returns {object} an object containing the opcode info necessary to deserialize the block
+ * @property {string} extendedOpcode - the full opcode for this block, like 'myExtension_this_that'
+ * @property {string} extensionId - the extension/category for this block, like 'myExtension'. May be an empty string.
+ */
+const getExtensionAndOpcode = function (blockJSON) {
+    const extendedOpcode = Array.isArray(blockJSON) ? primitiveIdMap[blockJSON[0]] : blockJSON.opcode;
+    const firstSplit = extendedOpcode.indexOf('_');
+
+    // note: if we ever need the simple / non-extended opcode, it's just extendedOpcode.substring(firstSplit + 1)
+    return {
+        extendedOpcode,
+        extensionId: extendedOpcode.substring(0, firstSplit)
```

Comment:
Elsewhere we use `StringUtil.splitFirst` to get this info, might be good to use that here for consistency.